components/event-publisher-proxy/pkg/sender/eventmesh/eventmesh.go

```diff
@@ -36,47 +37,43 @@ func (s *Sender) URL() string {
 	return s.Target
 }
 
-type HTTPPublishResult struct {
-	Status int
-	Body   []byte
-}
-
-func (h HTTPPublishResult) HTTPStatus() int {
-	return h.Status
-}
-
-func (h HTTPPublishResult) ResponseBody() []byte {
-	return h.Body
-}
-
 func (s *Sender) Checker() *health.ConfigurableChecker {
 	return &health.ConfigurableChecker{}
 }
 
-func (s *Sender) Send(ctx context.Context, event *cev2event.Event) (sender.PublishResult, error) {
+func (s *Sender) Send(ctx context.Context, event *cev2event.Event) sender.PublishError {
 	request, err := s.NewRequestWithTarget(ctx, s.Target)
 	if err != nil {
-		return nil, err
+		e := common.ErrInternalBackendError
+		e.Wrap(err)
+		return e
 	}
 
 	message := binding.ToMessage(event)
 	defer func() { _ = message.Finish(nil) }()
 
 	err = cloudevents.WriteRequestWithHeaders(ctx, message, request, additionalHeaders)
 	if err != nil {
-		return nil, err
+		e := common.ErrInternalBackendError
+		e.Wrap(err)
+		return e
 	}
 
 	resp, err := s.Client.Do(request)
 	if err != nil {
-		return nil, err
+		e := common.ErrInternalBackendError
+		e.Wrap(err)
+		return e
+	}
+	if resp.StatusCode >= http.StatusOK && resp.StatusCode < http.StatusMultipleChoices {
+		return nil
 	}
 	body, err := io.ReadAll(resp.Body)
 	defer func() { _ = resp.Body.Close() }()
 	if err != nil {
-		return nil, err
+		return common.ErrInternalBackendError
```

Comment:
should we log the `err` object here? or wrap it with the returned error?

Reply:
yeah, logging is probably a good idea